src/__tests__/element-queries.js

```diff
@@ -212,6 +212,36 @@ test('can get sibling elements with aria-labelledby attrib ute', () => {
   expect(result[0].id).toBe('icon')
 })
 
+test('can filter results of label query based on selector', () => {
+  const {getAllByLabelText} = render(`
+    <div>
+      <label id="label1" for="input1">
+        Test Label
+        <input id="input2" />
+      </label>
+      <input id="input1" class="fancy-input" />
+      <span aria-labelledby="label1">Some hint text</span>
+    </div>
+  `)
+
+  const result = getAllByLabelText('Test Label', {selector: '.fancy-input'})
+  expect(result).toHaveLength(1)
+  expect(result[0].id).toBe('input1')
+})
+
+test('can find input when label text is in a span', () => {
+  const {getAllByLabelText} = render(`
+    <label>
+      <span>Test Label</span>
+      <input id="input1" />
+    </label>
+  `)
+
+  const result = getAllByLabelText('Test Label', {selector: 'input'})
+  expect(result).toHaveLength(1)
+  expect(result[0].id).toBe('input1')
+})
+
```

Comment:
Could you add another test for the situation I found myself in where we had two elements that had the same label text?

```javascript
  const {getByLabelText} = render(`
    <label>
      Test Label
      <input />
    </label>
    <label>
      Test Label
      <textarea></textarea>
    </label>
  `)
  // select the input
```

Thanks!